docs/azure/sdk/protocol-convenience-methods.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Understand Azure SDK client library method types
+description: Learn about the key differences between Azure SDK client library protocol and convenience methods
+ms.topic: conceptual
+ms.custom: devx-track-dotnet, engagement-fy23, devx-track-arm-template
+ms.date: 06/24/2024
+---
+
+# Azure SDK for .NET protocol and convenience methods overview
+
+The Azure SDK client libraries provide an interface to Azure service REST APIs by translating method calls into underlying HTTP requests. In this article, you'll learn about the different types of methods exposed by the client libraries and explore their implementation patterns.
```

Comment:
This is super nit-picky of me to say, but we have some Azure SDK clients that talk to non-HTTP services, which I think are not REST based.  But those ones don't have protocol methods (at least, at present we don't have a story around that).  I'm happy to let you rework to address however you think is best given that, but one suggestion might be to say something like "interfaces to Azure services by translating method calls into messages sent via the service protocol.  For REST API services, this means sending HTTP requests and turning the responses into runtime types."  ... or something?